docs/tutorials/csharp/basic_csharp.md

```diff
@@ -1,36 +1,150 @@
 ---
-nav_exclude: true 
+title:  Basic C# Tutorial
+description: Basic usage of C# API
+parent: Inference with C#
+grand_parent: Tutorials
+has_children: false
+nav_order: 1
 ---
+
+
 # C# Tutorial: Basic
 
-Here is simple tutorial for getting started with running inference on an existing ONNX model for a given input data. The model is typically trained using any of the well-known training frameworks and exported into the ONNX format. 
+Here is a simple tutorial for getting started with running inference on an existing ONNX model for a given input data.
+The model is typically trained using any of the well-known training frameworks and then exported into the ONNX format. 
+
+Note, that the following classed `NamedOnnxValue`, `DisposableNamedOnnxValue`, `FixedBufferOnnxValue` are going
+to be deprecated in the future. They are not recommended for new code.
+
+The new `OrtValue` based API is the recommended approach. The `OrtValue` API generates less garbage and is more performant.
+Some scenarios indicated 4x performance improvement over the previous API and significantly less garbage.
+It provides uniform access to data via `ReadOnlySpan<T>` and `Span<T>` structures, regardless of its location, managed or unmanaged.
+
+`DenseTensor` class can still be used for multi-dimensional access to the data since the new `Span` based API feature
+only a 1-D index. However, some reported a slow performance when using `DenseTensor` class multi-dimensional access.
+One can then create an OrtValue on top of the tensors data. 
+
+`ShapeUtils` class provides some help to deal with multi-dimensional indices for OrtValues.
+
+`OrtValue` based API provides direct native memory access in a type safe manner using `ReadOnlySpan<T>` and `Span<T>` stack bases structures.
+OrtValue is a universal container that can hold different ONNX types, such as tensors, maps, and sequences.
+It always existed in the onnxruntime library, but was not exposed in the C# API.
+
+As before, `OrtValues` can be created directly on top of the managed `unmanaged` (struct based blittable types)  arrays.
+Read MS documentation on `blittable` data types. onnxruntime C# API allows use of managed buffers for input or output.
```

Comment:
Could add a link to https://learn.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types